models/schema.prisma

```diff
@@ -13,4 +13,148 @@ model Session {
+    role            Role              @default(USER)
+    isActive        Boolean
+    jobTitle        String
+    position        Positon           @relation(fields: [positionId], references: [id])
+    positionId      String
+    city            String
+    imageUrl        String
+    linkedIn        String
+    github          String
+    post            Post[]
+    savedPost       SavedPost[]
+    comment         Comment[]
+    reportedContent ReportedContent[]
+    log             Log[]
+    likes           Likes[]
+}
+
+model Post {
+    id            String            @id @default(uuid())
+    user          User              @relation(fields: [userId], references: [id])
+    userId        String            @unique
+    companyName   String
+    city          String
+    jobTitle      String
+    position      String
+    jobAdUrl      String
+    createdDate   DateTime          @default(now())
+    updatedDate   DateTime          @updatedAt
+    description   String
+    status        Status            @default(DRAFT)
+    category      Category
+    postTags      PostTag[]
+    report        ReportedContent[]
+    SavedPost     SavedPost[]
+    targetPost    Comment[]         @relation("targetPost")
+    targetComment Comment[]         @relation("targetComment")
+    likes         Likes[]
+}
+
+model PostTag {
+    id     String @id @default(uuid())
+    post   Post   @relation(fields: [postId], references: [id])
+    postId String
+    tag    Tag    @relation(fields: [tagId], references: [id])
+    tagId  String
+}
+
+model SavedPost {
+    id     String @id @default(uuid())
+    user   User   @relation(fields: [userId], references: [id])
+    userId String
+    post   Post   @relation(fields: [postId], references: [id])
+    postId String
+}
+
+model Likes {
+    id          String @id @default(uuid())
+    likedUser   User   @relation(fields: [likedUserId], references: [id])
+    likedUserId String
+    likedPost   Post   @relation(fields: [likedPostId], references: [id])
+    likedPostId String
+}
+
+model Tag {
+    id      String    @id @default(uuid())
+    name    String
+    postTag PostTag[]
+}
+
+model Comment {
+    id              String            @id @default(uuid())
+    user            User              @relation(fields: [userId], references: [id])
+    userId          String
+    targetPost      Post?             @relation("targetPost", fields: [targetPostId], references: [id])
+    targetPostId    String?
+    targetCommant   Post?             @relation("targetComment", fields: [targetCommentId], references: [id])
+    targetCommentId String?
+    type            CommentType
+    content         String
+    createdDate     DateTime          @default(now())
+    updatedDate     DateTime?
+    reportedContent ReportedContent[]
+}
+
+model ReportedContent {
+    id                String      @id @default(uuid())
+    reportedUser      User        @relation(fields: [reportedUserId], references: [id])
```

Comment:
Change reportedUser to reporter and reportedUserId to reporterId - this is the Id for the person who reported the post, not the person who was reported. We can get the details of the person who was reported from the post or comment